File: src/backend/Application/GoalAggregate/Commands/EditActionItem/EditActionItemCommand.cs
using InternetSafetyPlan.Application.Base;
using MediatR;

namespace InternetSafetyPlan.Application.GoalAggregate.Commands;

public record EditActionItemCommand(Guid GoalId, Guid ActionItemId, string Name, string? Description = null, DateTime? DueDate = null) : ICommand<Unit>;

File: src/frontend/src/library/goalAggregate/commands/EditActionItemCommand.ts
export class EditActionItemCommand {
  public goalId: string;
  public actionItemId: string;
  public name: string;
  public description: string | null;
  public dueDate: string | null;

  constructor(goalId: string, actionItemId: string, name: string, description: string | null, dueDate: string | null) {
    this.goalId = goalId;
    this.actionItemId = actionItemId;
    this.name = name;
    this.description = description;
    this.dueDate = dueDate;
  }
}